src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1455,6 +1542,39 @@ public ValidateSetAttribute(params string[] validValues)
 
             _validValues = validValues;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the ValidateSetAttribute class.
+        /// Valid values is returned dynamically from a custom class implementing 'IValidateSetValuesGenerator' interface.
+        /// </summary>
+        /// <param name="valuesGeneratorType">class that implements the 'IValidateSetValuesGenerator' interface</param>
+        /// <exception cref="ArgumentException">for null arguments</exception>
+        public ValidateSetAttribute(Type valuesGeneratorType)
+        {
+            if (!typeof(IValidateSetValuesGenerator).IsAssignableFrom(valuesGeneratorType))
+            {
+                throw PSTraceSource.NewArgumentException("valuesGeneratorType");
+            }
+
+            validValuesGeneratorType = valuesGeneratorType;
```

Comment:
Why store the type? If we stored the instance, we only look up in `s_ValidValuesGeneratorCache` once per use of the attribute instead of once per invocation of the command using the attribute.